analysis/pfe_methods/unicode_range_data/slicing_strategy_loader.py

```diff
@@ -0,0 +1,78 @@
+"""Helper that can load slicing strategy protos by name."""
+
+import io
+import os
+
+from analysis.pfe_methods.unicode_range_data import slicing_strategy_pb2
+from fontTools import ttLib
+from google.protobuf import text_format
+
+SLICING_STRATEGY_DIR = "analysis/pfe_methods/unicode_range_data"
+
+CACHE = dict()
+
+
+def slicing_strategy_for_font(font_bytes):  # pylint: disable=unused-argument
+  """Determines which slicing strategy should be used for the given font."""
+
+  # Slicing strategy is picked by counting what % of the codepoints in the font
+  # are covered by each available strategy. Choose the strategy with the highest
+  # coverage.
+  codepoints = codepoints_in_font(font_bytes)
+  strategy_scores = dict()
+  for strategy_name in get_available_strategies():
+    strategy = load_slicing_strategy(strategy_name)
+    all_strategy_codepoints = set()
+    for subset in strategy:
+      all_strategy_codepoints.update(subset)
+
+    strategy_scores[strategy_name] = sum(
+        codepoint in all_strategy_codepoints for codepoint in codepoints)
+
+  return max(strategy_scores.keys(),
+             key=lambda strategy: strategy_scores[strategy])
+
+
+def get_available_strategies():
+  """Returns the names of all available strategies."""
+  return set(
+      f.replace(".textproto", "")
+      for f in os.listdir(SLICING_STRATEGY_DIR)
+      if os.path.isfile(os.path.join(SLICING_STRATEGY_DIR, f)) and
+      f.endswith(".textproto"))
+
+
+def load_slicing_strategy(strategy_name):
```

Comment:
can you just annotate @lru_cache? - https://docs.python.org/3/library/functools.html

Reply:
Done.